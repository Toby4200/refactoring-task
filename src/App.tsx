import Pagination from './components/Pagination';
import Select from './components/Select';
import Posts from './components/Posts';
import usePostsQuery from './hooks/usePostsQuery';
import Container from './templates/Container';
import Loader from './components/Loader';

const App = () => {
  const { data, loading, limit, page, handleLimitChange, handlePageChange } =
    usePostsQuery();

  return (
    <Container>
      {loading && <Loader />}

      <Select limit={limit} handleLimitChange={handleLimitChange} />

      {data ? <Posts posts={data} /> : <p>Posts Loading...</p>}

      <Pagination page={page} handlePageChange={handlePageChange} />
    </Container>
  );
};

export default App;
