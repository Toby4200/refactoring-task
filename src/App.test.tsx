import App from './App';
import { createRoot } from 'react-dom/client';
import { render } from '@testing-library/react';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const root = createRoot(div);

  root.render(<App />);
});

it('matches snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
