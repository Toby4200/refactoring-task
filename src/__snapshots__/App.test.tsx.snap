// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    style="color: blue; box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.1); padding: 1rem;"
  >
    <span
      style="font-size: 2rem; font-weight: bold;"
    >
      Posts App
    </span>
  </div>
  <div
    style="position: relative; padding: 20px; max-width: 1200px; margin: 0px auto;"
  >
    <div
      style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; background-color: rgba(0, 0, 0, 0.5); z-index: 9998; cursor: wait;"
    />
    <div
      style="position: fixed; top: 50%; left: 50%; transform: translate(-50%, -50%); width: 40px; height: 40px; border: 3px solid #f3f3f3; border-top: 3px solid #3498db; border-radius: 50%; animation: spin 1s linear infinite; z-index: 9999;"
    />
    <label
      for="recordsPerPage"
    >
      Records per page: 
    </label>
    <select
      id="recordsPerPage"
      style="padding: 5px 10px; margin-bottom: 2rem;"
    >
      <option
        value="5"
      >
        5
      </option>
      <option
        value="10"
      >
        10
      </option>
      <option
        value="15"
      >
        15
      </option>
      <option
        value="20"
      >
        20
      </option>
    </select>
    <p>
      Posts Loading...
    </p>
    <p>
      Current page: 
      1
    </p>
    <button
      style="background-color: green; color: white; padding: 5px 10px;"
    >
      Previous
    </button>
    <button
      style="background-color: green; color: white; padding: 5px 10px;"
    >
      Next
    </button>
  </div>
</div>
`;
